cmd/gocritic/main.go

```diff
@@ -176,11 +185,20 @@ func (l *linter) checkFile(f *ast.File) {
 				}
 			}()
 
-			for _, warn := range c.Check(f) {
-				pos := l.ctx.FileSet.Position(warn.Node.Pos())
-				log.Printf("%s: %s: %v\n", pos, c.Rule, warn.Text)
+			if !l.ctx.QuickFix {
+				for _, warn := range c.Check(f) {
+					pos := l.ctx.FileSet.Position(warn.Node.Pos())
+					log.Printf("%s: %s: %v\n", pos, c.Rule, warn.Text)
+				}
```

Comment:
You don't call `c.Check` if `l.ctx.QuickFix` is true.
So, with quickfix, no checkers are executed.
If you will actually execute them, updated AST will be printed, as expected (I've verified that).

Reply:
Oh, god... Stupid mistake :D Thanks for finding it.